### PR TITLE
metrics: Instrumented provider http clients

### LINKED ISSUE #206: Expose and configure instrumented client metrics
look at http instrumented go pkg, and try to use that with our client used in the DNS Providers.

### PR DESCRIPTION
**Pull request by mikenairn**
closes #206 

Updates all dns provders to use a http client wrapped with prometheus client middleware to add generic http metrics for all requests made to external dns provider platforms (aws, gcp, azure).

Metrics produced are:

```
# HELP http_client_in_flight_requests A gauge of in-flight requests for the wrapped client.
# TYPE http_client_in_flight_requests gauge
http_client_in_flight_requests{client="aws"} 0
# HELP http_client_request_latency_seconds A histogram of request latencies.
# TYPE http_client_request_latency_seconds histogram
http_client_request_latency_seconds_bucket{client="aws",code="200",method="get",le="0.005"} 0
...
http_client_request_latency_seconds_bucket{client="aws",code="200",method="get",le="+Inf"} 157
http_client_request_latency_seconds_sum{client="aws",code="200",method="get"} 31.374643888999994
http_client_request_latency_seconds_count{client="aws",code="200",method="get"} 157
...
# HELP http_client_requests_total A counter for requests from the wrapped client.
# TYPE http_client_requests_total counter
http_client_requests_total{client="aws",code="200",method="delete"} 3
http_client_requests_total{client="aws",code="200",method="get"} 157
http_client_requests_total{client="aws",code="200",method="post"} 30
http_client_requests_total{client="aws",code="201",method="post"} 3
http_client_requests_total{client="aws",code="400",method="post"} 14
http_client_requests_total{client="aws",code="403",method="get"} 12
http_client_requests_total{client="aws",code="404",method="get"} 2
```

Additional changes:
* Removes the dependency on `github.com/linki/instrumented_http` in favour of just using the[ prometheus http middleware](https://github.com/prometheus/client_golang/blob/86f34968685552f0993747630ba5be1ab2f3e3cd/prometheus/promhttp/instrument_client.go).
* Removes the old, unused, AWS instrumented client code.

Metrics output from an e2e test run for all providers(aws, google,azure) against a single instance: https://gist.github.com/mikenairn/4fc6f3437ea4af07e55eb3a018af78b9

Related [PR](https://github.com/Kuadrant/dns-operator/pull/246) to archive dns operator metrics at end of e2e run.

### CODE REVIEW

**Comment by mikenairn** [comment]
@david-martin Wouldn't mind your thoughts on this, do the [http metrics](https://gist.github.com/mikenairn/4fc6f3437ea4af07e55eb3a018af78b9#file-e2e-test-run-metrics-all-providers-L183-L446) being outputted look reasonable to you?

**Comment by maleck13** [comment]
@guicassolato does authorino have this type of client metrics for requests to external sources?

File: internal/metrics/http.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		prometheus.HistogramOpts{
+			Name:    "http_client_request_latency_seconds",
+			Help:    "A histogram of request latencies.",
+			Buckets: prometheus.DefBuckets,
+		},
+		[]string{"client", "code", "method"},
+	)
+	clientInFlight = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "http_client_in_flight_requests",
```
Comment (maleck13) [inline internal/metrics/http.go:30]: This is an interesting one. So you can see how many requests were in flight for a given time period. 👍

**Review by maleck13 — APPROVED** [review]
This a great set of metrics IMO

**Pull request merged**